### PR TITLE
Update dependencies

### PR DESCRIPTION
**Pull request by rock3r**
## Problem

We have a bunch of outdated dependencies

## Solution

Update (almost) all of them — see commits history for details.

What we did not update is the `Gradle Static Analysis Plugin` because, although it supports Detekt and Android Lint starting from 0.5.x, a trivial porting seemed to uncover some issues. That plugin will thus be updated separately.

### Test(s) added 

No but builds and it's green.

### Paired with 

Nobody

### CODE REVIEW

File: app/src/main/java/net/squanchy/eventdetails/widget/EventDetailsCoordinatorLayout.kt [hunk 1]
```diff
@@ -27,8 +27,7 @@ class EventDetailsCoordinatorLayout @JvmOverloads constructor(
             favoriteFab.setImageResource(
                     if (event.favorited)
                         R.drawable.ic_favorite_filled
-                    else
-                        R.drawable.ic_favorite_empty
+                    else R.drawable.ic_favorite_empty
```
Comment (rock3r) [inline app/src/main/java/net/squanchy/eventdetails/widget/EventDetailsCoordinatorLayout.kt:30]: KtLint did this

File: app/build.gradle [hunk 2]
```diff
@@ -54,7 +54,7 @@ android {
         versionName project.VERSION_NAME
 
         manifestPlaceholders += [
-                fabricApiKey: (buildProperties.secrets['fabricApiKey'] | buildProperties.env['FABRIC_API_KEY']).string
+            fabricApiKey: (buildProperties.secrets['fabricApiKey'] | buildProperties.env['FABRIC_API_KEY']).string
```
Comment (rock3r) [inline app/build.gradle:57]: `cmd-alt-l`'d

File: app/src/main/java/net/squanchy/imageloader/GlideImageLoader.kt [hunk 3]
```diff
@@ -18,8 +18,7 @@ class GlideImageLoader(private val requestManager: RequestManager, private val f
     override fun load(url: String): ImageRequest {
         return if (url.startsWith(FIREBASE_URL_SCHEMA))
             load(firebaseStorage.getReferenceFromUrl(url))
-        else
-            GlideImageRequest(requestManager.load(url))
+        else GlideImageRequest(requestManager.load(url))
```
Comment (rock3r) [inline app/src/main/java/net/squanchy/imageloader/GlideImageLoader.kt:21]: KtLint did this

**Comment by rock3r** [comment]
Note that the bulk of changes derives, once again, from the merging of the Detekt ruleset changes in the latest release (there are new rules, and rules are now alphabetically sorted, and some style rules have been moved to the naming group).

**Review by quiro91 — APPROVED** [review]

**Pull request merged**